webapp/src/js/enketo/widgets/phone-widget.js

```diff
@@ -44,49 +42,38 @@ FormModel.prototype.types.tel = {
 /**
    * Allows validated phonenumber entry.
    *
-   * @constructor
-   * @param {Element} element [description]
-   * @param {(boolean|{touch: boolean, repeat: boolean})} options options
-   * @param {*=} e     event
+   * @extends Widget
    */
+class PhoneWidget extends Widget {
+  constructor( element, options, Settings = window.CHTCore.Settings ) {
+    super(element, options);
 
-function PhoneWidget( element, options, Settings ) {
-  this.namespace = pluginName;
-  Widget.call( this, element, options );
-  if ( !Settings ) {
-    Settings = window.CHTCore.Settings;
-  }
-  this._init( Settings );
-}
-
-//copy the prototype functions from the Widget super class
-PhoneWidget.prototype = Object.create( Widget.prototype );
-
-//ensure the constructor is the new one
-PhoneWidget.prototype.constructor = PhoneWidget;
+    const $input = $( this.element );
 
-PhoneWidget.prototype._init = function( Settings ) {
-  const $input = $( this.element );
+    // Add a proxy input field, which will send its input, formatted, to the real input field.
+    // TODO(estellecomment): format the visible field onBlur to user-friendly format.
+    const $proxyInput = $input.clone();
+    $proxyInput.addClass('ignore');
+    $proxyInput.removeAttr('data-relevant');
+    $proxyInput.removeAttr('name');
+    $input.before( $proxyInput );
+    $proxyInput.val( $input.val() );
 
-  // Add a proxy input field, which will send its input, formatted, to the real input field.
-  // TODO(estellecomment): format the visible field onBlur to user-friendly format.
-  const $proxyInput = $input.clone();
-  $proxyInput.addClass('ignore');
-  $proxyInput.removeAttr('data-relevant');
-  $proxyInput.removeAttr('name');
-  $input.before( $proxyInput );
-  $proxyInput.val( $input.val() );
+    $input.hide();
 
-  $input.hide();
+    // TODO(estellecomment): move this to a catch clause, when settings aren't found.
+    formatAndCopy( $proxyInput, $input, {} );
 
-  // TODO(estellecomment): move this to a catch clause, when settings aren't found.
-  formatAndCopy( $proxyInput, $input, {} );
+    return Settings.get()
+      .then( function( settings ) {
+        formatAndCopy( $proxyInput, $input, settings );
+      } );
+  }
 
-  this.builtPromise = Settings.get()
-    .then( function( settings ) {
-      formatAndCopy( $proxyInput, $input, settings );
-    } );
-};
+  static get selector() {
+    return 'input[type="tel"][data-type-xml="tel"]';
```

Comment:
This is kinda dumb (the whole numbers -> tel). Please add a comment about why this should exist, otherwise this will be a pretty hard question to answer years from now.